massa-hash/src/hash_xof.rs

```diff
@@ -0,0 +1,57 @@
+use std::ops::{BitXor, BitXorAssign};
+
+use crate::{Hash, XOF_HASH_SIZE_BYTES};
+
+/// Extended Hash
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct HashXof(pub [u8; XOF_HASH_SIZE_BYTES]);
+
+impl HashXof {
```

Comment:
if we go for genericity, we should have the size be a const template parameter :)

Reply:
https://practice.rs/generics-traits/const-generics.html